aws/sagemaker-distribution
# fix: add tensorflow and pytorch CUDA version tests for GPU image build

*Issue #, if available:*
In 1.8.0 GPU image, we installed tensorflow-cpu version by mistake

*Description of changes:*
To get rid of such issue, add a unit test to validate tensorflow and pytorch cuda version is installed in GPU image

*Test:*
1. Build v1.9.0 locally
2. Run test against GPU image, it succeed
3. Run test against CPU image, test failed:
```
Exception: TensorFlow is installed without CUDA support for GPU image build.
``` 
4. Run first Pytorch validation test against CPU image, test failed:
```
Exception: Pytorch is installed without CUDA support for GPU image build.
```
5. Run second Pytorch validation test against CPU image, test failed:
```
Exception: Pytorch CUDA is not working in current environment. Make sure to execute this test case in GPU environment if you are not
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
